### PR TITLE
adding tag option in qq upload

### LINKED ISSUE #505: Add tags to qq-upload

### PR DESCRIPTION
**Pull request by scarrazza**
Closes #505.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.
- [ ] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [ ] Qibo: `master`
    - [ ] Qibolab: `main`
    - [ ] Qibolab_platforms_qrc: `main`

### CODE REVIEW

**Review by andrea-pasquale — COMMENTED** [review]
Thanks @scarrazza, the code looks good to me.
I tested it, however I cannot  see the report in http://login.qrccluster.com:9000/.
I remember that also last time I was having some issues, maybe we need to restart the server(?)
Also @GabrielePalazzo was looking to test it but his SSH key is not in the qibocal user, is there a way in which we could automate this process or should he just send to you his ssh-key?
Last thing, I saw that some reports are not correctly rendered in the web page
Screenshot from 2023-09-21 12-03-36
At this point maybe it is better to just clean this folder and start from scratch.

File: src/qibocal/cli/upload.py [hunk 1]
```diff
@@ -19,7 +23,13 @@
 ROOT_URL = "http://login.qrccluster.com:9000/"
 
 
-def upload_report(path: pathlib.Path):
+def upload_report(path: pathlib.Path, tag: str):
+    # load meta and update tag
+    if tag is not None:
+        meta = yaml.safe_load((path / META).read_text())
+        meta["tag"] = tag
+        (path / META).write_text(json.dumps(meta, indent=4))
```
[the remark this answers is not in the document]
Comment (scarrazza) [reply src/qibocal/cli/upload.py:31]: I believe this is for readability, we are already doing the same for the acquire, etc...

[the remark this answers is not in the document]
Comment (andrea-pasquale) [reply src/qibocal/cli/upload.py:31]: It is fine then.

**Comment by scarrazza** [comment]
@andrea-pasquale thanks, I did not update the server yet, but if this approach looks okay, I will do asap.

**Comment by scarrazza** [comment]
@GabrielePalazzo @andrea-pasquale the server has been just updated.

**Comment by andrea-pasquale** [comment]
> @GabrielePalazzo @andrea-pasquale the server has been just updated.

I see that the server has been updated, however I don't see yet my report uploaded just now.

**Comment by scarrazza** [comment]
Did you change meta.yaml in meta.json? Do you confirm that your last report `_tHWAo02Tjitw_iFxTTcPQ==` does not have any tag value?

**Comment by andrea-pasquale** [comment]
> Did you change meta.yaml in meta.json?

Yes, I did. I'm sorry I forgot that also the server part relies on this file.



> Do you confirm that your last report `_tHWAo02Tjitw_iFxTTcPQ==` does not have any tag value?

I can confirm that there is not tag.

**Comment by scarrazza** [comment]
Please check now.

File: src/qibocal/cli/upload.py [hunk 2]
```diff
@@ -19,7 +23,13 @@
 ROOT_URL = "http://login.qrccluster.com:9000/"
 
 
-def upload_report(path: pathlib.Path):
+def upload_report(path: pathlib.Path, tag: str):
+    # load meta and update tag
+    if tag is not None:
```
Comment (GabrielePalazzo) [inline src/qibocal/cli/upload.py:28]: In case tag is left as default (`None`), do we want to leave the tag field empty?

**Comment by andrea-pasquale** [comment]
> Please check now.

Now everything works as expected, thanks @scarrazza.

**Review by andrea-pasquale — APPROVED** [review]

**Review by GabrielePalazzo — COMMENTED** [review]
Looks good to me too.

File: src/qibocal/cli/upload.py (hunk 2 above)
Comment (scarrazza) [reply src/qibocal/cli/upload.py:28]: I am open for suggestions. At the moment the server replaces empty tags with `-`.

Comment (GabrielePalazzo) [reply src/qibocal/cli/upload.py:28]: Indeed. Each report has its own unique identifier, so I suppose if no tag is specified we can leave `-`.

**Review by GabrielePalazzo — APPROVED** [review]

**Pull request merged**